corecfg/proxy.go

```diff
@@ -29,9 +29,9 @@ import (
 )
 
 var proxyConfigKeys = map[string]bool{
-	"http":  true,
-	"https": true,
-	"ftp":   true,
+	"http_proxy":  true,
```

Comment:
This is properly fixed and tested in https://github.com/snapcore/snapd/pull/3754 - maybe we should focus on this one instead than this one here? Unfortunately removing this change will break the tests (because the bugfix in here exposes a bug in the proxy handling)